Orange/widgets/visualize/owsilhouetteplot.py

```diff
@@ -574,11 +574,11 @@ def clear(self):
 
     def __setup(self):
         # Setup the subwidgets/groups/layout
-        smax = max((numpy.max(g.scores) for g in self.__groups
+        smax = max((numpy.max(numpy.nan_to_num(g.scores)) for g in self.__groups
```

Comment:
Why not `numpy.nanmax` (here) and `numpy.nanmin` (below)? Plus, set it to 1 or -1 if the length is zero or all values are NaN.